### PR TITLE
Don't load multiple folders at once

### LINKED ISSUE #467: Multiple folders get loaded when switching fast
As @wurstchristoph mentioned in issue #440:

>  if you switch folders that fast, the message column is loaded for each folder you clicked (ajax responses coming in). This looks odd and should be fixed too

When another folder is clicked, the previous ajax call should probably be cancelled right away.

@DeepDiver1975 @zinks- something for you? :)

### PR DESCRIPTION
**Pull request by zinks-**
Reference to the currently loading request is kept in Mail.State and aborted if another folder is loaded.
TextStatus from the $.ajax error handler is trickled down to prevent handling it as a real error.

EDIT: changed the PR name to something more explicit ;)

### CODE REVIEW

**Comment by zinks-** [comment]
@jancborchardt @DeepDiver1975 @wurstchristoph Please review

**Comment by DeepDiver1975** [comment]
unit tests on travis will fail because the encrypted test credentials cannot be decrypted

File: js/mail.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ var Mail = {
 		currentFolderId: null,
 		currentAccountId: null,
 		currentMessageId: null,
+                currentlyLoading: null,
```
Comment (wurstchristoph) [inline js/mail.js]: you used spaces here, they should be replaced by tabs ;-)

**Comment by jancborchardt** [comment]
Otherwise it looks like it’s fixed for me. :+1: 

@DeepDiver1975 would rebasing fix the unit tests?

File: js/mail.js (hunk 1 above)
Comment (zinks-) [inline js/mail.js]: We should have a .editorconfig :D

Comment (jancborchardt) [inline js/mail.js]: So, can you fix it? :) Then we can merge it I guess. @wurstchristoph did you test it and also works for you?

**Comment by jancborchardt** [comment]
@zinks- can you replace the spaces with tabs please? ;)

File: js/mail.js (hunk 1 above)
Comment (wurstchristoph) [inline js/mail.js]: Yes, i did and it works :+1:

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 would rebasing fix the unit tests?

no - the "issue" is that this PR is coming from a different repo - the secured data can then not be read by travis because the private key is within our repo - this is explained somewhere in the travis docs ....

**Comment by zinks-** [comment]
Messed up my commit history, but the result should me mergeable (had a merge conflict) and has only tabs for indent.
If you can tell me what git commands to use to clean the history up I will gladly do it.

**Comment by jancborchardt** [comment]
@zinks- something with "squash". That allows you to put all the changes you made into one commit.

**Comment by zinks-** [comment]
@jancborchardt thanks, it's done.

**Comment by jancborchardt** [comment]
@zinks- nice! Merged. :)

**Pull request merged**